### DISCUSSION TITLE
configurazione core-mosquitto

### DISCUSSION THREAD

**Discussion by loregcl** [indomus/forum#2363] [HUB: Home Assistant]
Ho installato HomeAssistantOS e sto cercando di configurare l'addon mosquitto. ho aggiunto l'utente mqtt_user, nel login ho inserito le stesse credenziali dell'utente, tutto si avvia correttamente senza errori.
Ho aggiunto l'integrazione mqtt e inserito nel nome del brocker alternativamente "localhost", core-mosquitto, l'IP locale; per le credenziali le stesse inserite nell'utente e nella configurazione dell'addon.
Per verificare se tutto funziona ho installato l'integrazione tasmota e ho configurato un paio di prodotti per parlare con il broker.
ho questo errore dal registro di debug e non riesco a capire cosa succeda:

```
2023-02-26 19:16:24: New connection from 172.30.32.2:41508 on port 1883.
2023-02-26 19:16:24: Client <unknown> closed its connection.
2023-02-26 19:17:01: New connection from <IPtasmota>:51685 on port 1883.
error: received null username or password for unpwd check
2023-02-26 19:17:01: Sending CONNACK to <IPtasmota> (0, 5)
2023-02-26 19:17:01: Client <unknown> disconnected, not authorised.
2023-02-26 19:17:10: Received PINGREQ from 5NWSCEgnYvCu3buBe4gs4U
2023-02-26 19:17:10: Sending PINGRESP to 5NWSCEgnYvCu3buBe4gs4U
```
l'IP 172.30.32.2 dovrebbe essere il supervisor... neanche lui riesce a entrare.

ho notato che nella configurazione nonostante io inserisca gli apici su username e pwd, salvando la configurazione gli apici scompaiono su username.

Su homeassitant core funzionava tutto in maniera così semplice, non mi spiego cosa stia succedendo.

grazie a tutti in anticipo.

**Comment by snakuzzo** [comment]
Se non hai bisogno di certificati o roba simile, la configurazione deve essere questa (attento alla sezione Customize)

![image](https://user-images.githubusercontent.com/7205371/221438078-4ebb4a2c-8f49-44fa-b270-4cccb52d9b40.png)

**Reply by loregcl** [reply]
Questo il mio file di configurazione dell'addon:

```yaml
logins:
  - username: user
    password: "pwd"
require_certificate: false
certfile: fullchain.pem
keyfile: privkey.pem
customize:
  active: false
  folder: mosquitto
debug: true
```

ricevo sempre lo stesso errore da tutti i clients che provano a connettersi...

```
2023-02-27 07:47:55: New connection from <IPtasmota01>:57154 on port 1883.
error: received null username or password for unpwd check
2023-02-27 07:47:55: Sending CONNACK to <IPtasmota01> (0, 5)
2023-02-27 07:47:55: Client <unknown> disconnected, not authorised.
2023-02-27 07:48:20: New connection from <IPtasmota02>:55344 on port 1883.
error: received null username or password for unpwd check
2023-02-27 07:48:20: Sending CONNACK to <IPtasmota02> (0, 5)
2023-02-27 07:48:20: Client <unknown> disconnected, not authorised.
2023-02-27 07:48:27: New connection from 172.30.32.2:51158 on port 1883.
2023-02-27 07:48:27: Client <unknown> closed its connection.
2023-02-27 07:48:51: Received PINGREQ from 3QqmAU7X4G7bk86l2bsGz8
2023-02-27 07:48:51: Sending PINGRESP to 3QqmAU7X4G7bk86l2bsGz8
2023-02-27 07:49:52: Received PINGREQ from 3QqmAU7X4G7bk86l2bsGz8
2023-02-27 07:49:52: Sending PINGRESP to 3QqmAU7X4G7bk86l2bsGz8
2023-02-27 07:49:56: New connection from <IPtasmota01>:60444 on port 1883.
error: received null username or password for unpwd check
```
non riesco proprio a capire cosa stia succedendo...

**Reply by KC-inDomus** [reply]
a me pare che sia un client misconfigurato, IPtasmota02, a tentare di autenticarsi senza riuscirci

**Reply by loregcl** [reply]
Se fosse così almeno 172.30.32.2 dovrebbe entrare...
Con homeassistant core i client funzionavano, ho solo cambiato l'IP del brocker nella loro configurazione (metto uno screenshot perchè non so come fare altrimenti):

![parametri tasmota](https://user-images.githubusercontent.com/77339969/221518462-d826b80c-d399-40c2-978b-1aa25df71a38.jpg)

**Reply by snakuzzo** [reply]
`error: received null username or password for unpwd check`

**Reply by loregcl** [reply]
In realtà nei due tasmota che ho scelto per provare il servizio i campi sono compilati, ma lo stesso problema c'era sul servizio del 172.30.32.2.
Ho provato a scegliere il protocollo 3.1 al posto 3.1.1 che era impostato di default nell'integrazione MQTT e adesso il topic di homeassistant viene pubblicato.
L'ho testato con HiveMQ.
Ho provato a lanciare dei topic di prova dall'integrazione MQTT e vengono pubblicati correttamente.
A questo punto mancano solo i tasmota.
Può essere un problema di protocollo MQTT anche per quelli? altrimenti mi rimane solo di resettarli per capire se per qualche ragione strana il menù di configurazione del tasmota non aggiorna le credenziali di username e pwd quando cambi l'indirizzo del brocker...

**Reply by loregcl** [reply]
Ho resettato i device tasmota e niente da fare…
Ho aggiunto un nuovo utente, cambiato le credenziali nei tasmota e niente.
Ho forzato da console il SetOption19 a OFF come scritto della documentazione tasmota e niente.
Ho un altro device mqtt che viene letto  correttamente dall’integrazione mqtt.
Alla fine preso dalla disperazione ho cambiato la password e ho messo solo lettere mentre prima erano solo numeri e di colpo ha funzionato.

Credo a questo punto che il fw tasmota non formatti correttamente il campo pwd perché il dispositivo che ha sempre funzionato aveva una pwd composta solo da numeri.

Alla fine comunque ho risolto.

grazie per l’attenzione

**Reply by snakuzzo** [reply]
Comunque `SetOption19` non c'entra niente in tutto ciò.
Quello non serve al collegamento al broker MQTT bensì alla scelta se prevedere l'aggiunta automatica ad HA tramite autodiscovery integrazione MQTT (`SO19 1`) oppure tramite autodiscovery integrazione Tasmota ( (`SO19 0` che è il default).

**Reply by loregcl** [reply]
Sì sì, è che nella disperazione del momento le ho provate proprio tutte.
La questione dei numeri formattati male nel campo pwd da tasmota credo sia la causa, anche perchè un altro device con le stesse credenziali entrava, mentre i tasmota no. Strano che non sia mai successo a nessun altro...

**Reply by snakuzzo** [reply]
Mi pare strano, ma comunque è strano pure mettere password solo numeriche eh :D
